### PR TITLE
Vault

### PR DESCRIPTION
**Pull request by yotamdaniel**
Add vault.sol and tests

Note: what do we do about all these node modules? tried to gitignore but then it messes up directory and hardhat - can give it another try with peace

### CODE REVIEW

File: contracts/Vault.sol [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+        if (IERC20(oUSDContractAdress).balanceOf(address(this)) < amount) {
+            revert("Not enough funds in vault");
+        }
+        // approve inside contract as "user" cant approve it. Discuss if needed
+        IERC20(oUSDContractAdress).approve(address(this), amount);
+        IERC20(oUSDContractAdress).transferFrom(address(this), to, amount);
+    }
+
+    //Note: we can get rid of this call and save a static copy of total supply which is updated
```
Comment (yotamdaniel) [inline contracts/Vault.sol:34]: Remove

File: contracts/Vault.sol [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        oUSDContractAdress = oUSDContract;
+    }
+
+    function depositFunds(address from, uint256 amount) public onlyOwner {
+        console.log("balance of user is: ", IERC20(oUSDContractAdress).balanceOf(from));
+        IERC20(oUSDContractAdress).transferFrom(from, address(this), amount);
+    }
+
+    function witdrawFunds(address to, uint256 amount) public onlyOwner {
+        console.log("witdrawFunds: balance of vault's address is: ",IERC20(oUSDContractAdress).balanceOf(address(this)));
```
Comment (yotamdaniel) [inline contracts/Vault.sol:24]: Remove all logs!

File: contracts/Vault.sol [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+        if (IERC20(oUSDContractAdress).balanceOf(address(this)) < amount) {
+            revert("Not enough funds in vault");
+        }
+        // approve inside contract as "user" cant approve it. Discuss if needed
```
Comment (yotamdaniel) [inline contracts/Vault.sol:29]: remove

File: test/vault.js [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        hardhatOusd = await ousd.deploy();
+        hardhatVault = await vault.deploy(hardhatOusd.address);
+
+        // setup funds for address2, ie test customer
+        await hardhatOusd.transfer(endUserAddr.address, minTransferAmount)
+    });
+
+    describe("Deposit", function () {
+        it("depositFunds from user should transfer funds to vault's address", async function () {
+            // remove
```
Comment (yotamdaniel) [inline test/vault.js:32]: remove

**Review by tomerXarchimedes — APPROVED** [review]
Approve - test

File: test/vault.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    let ousd;
+    let hardhatOusd;
+    let owner;
+    let endUserAddr;
+    let addrs;
+
+    // in realife,x*sqr(10,18) is eaqul to one coin 
+    const minTransferAmount = 1
+
+    beforeEach(async function () {
```
Comment (yotamdaniel) [inline test/vault.js:18]: suggestion : add which tests run and what is expected?

**Pull request closed without merging**